postgresql: use team

Ma27 opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->
Main motivation for this is that I'd like to get a feature-freeze ping: we have old stuff to remove and quite a bit of things ongoing here, so explicitly being part of the check-up process seems like a good thing.

Also added myself and wolfgangwalther to it.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Ma27 committed:
postgresql: use team

Main motivation for this is that I'd like to get a feature-freeze ping:
we have old stuff to remove and quite a bit of things ongoing here, so
explicitly being part of the check-up process seems like a good thing.

Also added myself and wolfgangwalther to it.

wolfgangwalther reviewed (commented):
Ah, thanks for taking that up. I wanted to add to the new OWNERS file anyway, but didn't get around to that, yet. Didn't know there was a postgres team already, cool!

Happy to be in it, ofc.

Ma27 commented on maintainers/team-list.nix:
```diff
@@ -845,9 +845,10 @@ with lib.maintainers;
   postgres = {
-    members = [ thoughtpolice ];
+    members = [ thoughtpolice ma27 wolfgangwalther ];
     scope = "Maintain the PostgreSQL package and plugins along with the NixOS module.";
```
Don't we need a GH group for that?
cc @infinisil 
whom to ping for that btw?

Ma27 replied:
I think I'd prefer to have a GH team with all members of the maintainer team.
@infinisil do you know if  these are kept in sync or what is the process for these groups?

infinisil replied:
Any committer can create a new team, but you'll have to ask in https://matrix.to/#/%23org_owners:nixos.org to have it be given Nixpkgs read/triage permission, which is required to be requested for reviews and OWNERS

Ma27 replied:
> but any committer can create a new team

Ohh, I just assumed that you'd need more permissions than I have and didn't check, sorry and thanks for pointing this out!

Ma27 commented:
@wolfgangwalther just created a team for postgres with all members in this branch.

What do we do with the other maintainers? It's not super-urgent to merge it now, but I'd like to not stall this because we're awaiting feedback.

Ma27 commented on ci/OWNERS:
```diff
@@ -226,10 +226,10 @@ pkgs/development/python-modules/buildcatrust/ @ajs124 @lukegb @mweinelt
+/pkgs/servers/sql/postgresql @NixOS/postgres
+/nixos/modules/services/databases/postgresql.md @NixOS/postgres
+/nixos/modules/services/databases/postgresql.nix @NixOS/postgres
+/nixos/tests/postgresql.nix @NixOS/postgres
```
Perhaps we want to move all postgresql-related tests into `nixos/tests/postgresql`? That way, we could also add the subtree here.

Ma27 commented:
Done.
Let's perform the restructuring of the tests in #352966 if that's OK for you.

Ma27 committed:
postgresql: remove danbst from maintainer list

Thanks a lot for all your work! If you ever come back to nixpkgs, feel
free to revert the commit, your return would be welcomed!

Ma27 commented on pkgs/servers/sql/postgresql/generic.nix:
```diff
@@ -320,7 +320,7 @@ let
       license     = licenses.postgresql;
       changelog   = "https://www.postgresql.org/docs/release/${finalAttrs.version}/";
-      maintainers = with maintainers; [ thoughtpolice danbst globin ivan ma27 wolfgangwalther ];
+      maintainers = with maintainers; [ globin ivan ] ++ teams.postgres.members;
```
Done.

Ma27 commented:
So, the team exists, I now need somebody to add nixpkgs in read-only mode to be added to the team:

>  You do not have sufficient permissions to add the nixpkgs repository to this team  

cc @Lassulus @zimbatm can one of you add nixpkgs (role: readonly) to https://github.com/orgs/NixOS/teams/postgres ?

thoughtpolice commented on pkgs/servers/sql/postgresql/generic.nix:
```diff
@@ -320,7 +320,7 @@ let
       license     = licenses.postgresql;
       changelog   = "https://www.postgresql.org/docs/release/${finalAttrs.version}/";
-      maintainers = with maintainers; [ thoughtpolice danbst globin ivan ma27 wolfgangwalther ];
+      maintainers = with maintainers; [ danbst globin ivan ] ++ teams.postgres.members;
```
I still do use postgres-with-nix behind the scenes and even did some commercial work on it last year, but mostly it's just because I'm spread very thin and don't have time to give it singular focus and many of the PRs require much more attention than I have available.

I'm happy to remain on the maintainers list as a point-of-contact for questions though.
